src/databricks/labs/ucx/source_code/linters/dfsa.py

```diff
@@ -0,0 +1,176 @@
+    DFSAPattern("abfs:/", []),
+    DFSAPattern("abfss:/", []),
+    DFSAPattern("hdfs:/", []),
+    # "/mnt/" is detected by the below pattern,
+    RootPattern("/", ["Volumes/", "Workspace/", "tmp/"]),
+]
+
+
+@dataclass
+class DFSA:
+    """A DFSA is a record describing a Direct File System Access"""
+
+    path: str
+
+
+@dataclass
+class DFSANode:
+    dfsa: DFSA
+    node: NodeNG
+
+
+class _DetectDfsaVisitor(TreeVisitor):
+    """
+    Visitor that detects file system paths in Python code and checks them
+    against a list of known deprecated paths.
+    """
+
+    def __init__(self, session_state: CurrentSessionState, allow_spark_duplicates: bool) -> None:
+        self._session_state = session_state
+        self._dfsa_nodes: list[DFSANode] = []
+        self._reported_locations: set[tuple[int, int]] = set()
+        self._allow_spark_duplicates = allow_spark_duplicates
+
+    def visit_call(self, node: Call):
+        for arg in node.args:
+            self._visit_arg(arg)
+
+    def _visit_arg(self, arg: NodeNG):
+        try:
+            for inferred in InferredValue.infer_from_node(arg, self._session_state):
+                if not inferred.is_inferred():
+                    logger.debug(f"Could not infer value of {arg.as_string()}")
+                    continue
+                self._check_str_constant(arg, inferred)
+        except InferenceError as e:
+            logger.debug(f"Could not infer value of {arg.as_string()}", exc_info=e)
+
+    def visit_const(self, node: Const):
+        # Constant strings yield Advisories
+        if isinstance(node.value, str):
+            self._check_str_constant(node, InferredValue([node]))
+
+    def _check_str_constant(self, source_node, inferred: InferredValue):
+        if self._already_reported(source_node, inferred):
+            return
+        # avoid duplicate advices that are reported by SparkSqlPyLinter
+        if Tree(source_node).is_from_module("spark") and not self._allow_spark_duplicates:
+            return
+        value = inferred.as_string()
+        if any(pattern.matches(value) for pattern in DFSA_PATTERNS):
+            self._dfsa_nodes.append(DFSANode(DFSA(value), source_node))
+            self._reported_locations.add((source_node.lineno, source_node.col_offset))
+
+    def _already_reported(self, source_node: NodeNG, inferred: InferredValue):
+        all_nodes = [source_node] + inferred.nodes
+        return any((node.lineno, node.col_offset) in self._reported_locations for node in all_nodes)
+
+    @property
+    def dfsa_nodes(self):
+        return self._dfsa_nodes
+
+
+class DfsaPyLinter(PythonLinter):
+
+    def __init__(self, session_state: CurrentSessionState, allow_spark_duplicates=False):
+        self._session_state = session_state
+        self._allow_spark_duplicates = allow_spark_duplicates
+
+    @staticmethod
+    def name() -> str:
+        """
+        Returns the name of the linter, for reporting etc
+        """
+        return 'dfsa-usage'
```

Comment:
where is it used?

Reply:
removed, not sure why it was there in the original dbfs linters...